src/components/CodeHighlight/highlightAngularComponent.js

```diff
@@ -0,0 +1,69 @@
+import astroHighlightCode from "@/components/CodeHighlight/astroHighlightCode.js";
+
+export async function highlightAngularComponent(
+  fileContent,
+  theme,
+  wrap,
+  fileExt
+) {
+  const templateCode = getAngularTemplateCode(fileContent);
+
+  if (!templateCode) return fileContent;
+
+  const componentWithEmptyTemplate = removeAngularTemplateContent(fileContent);
+
+  let templateCodeHighlighted = await astroHighlightCode({
+    code: templateCode,
+    lang: "html",
+    theme,
+    wrap,
+  });
+
+  let componentHighlighted = await astroHighlightCode({
+    code: componentWithEmptyTemplate,
+    lang: fileExt,
+    theme,
+    wrap,
+  });
+
+  return componentHighlighted.replace(
+    "template",
+    "template: `" + removeCodeWrapper(templateCodeHighlighted) + "`,"
+  );
+}
+
+function getAngularTemplateCode(fileContent) {
+  // regex to grab whats inside angular component template inside backticks
```

Comment:
Ok so we need to be careful on future Angular snippets keeping the backticks

Reply:
Yep, that would be a requirement.